examples/javascript/simple_sponsored_transaction.js

```diff
@@ -0,0 +1,97 @@
+  console.log(`Bob's address is: ${bobAddress}`);
+  console.log(`Sponser's address is: ${sponserAddress}`);
+
+  // Fund Alice and Sponser accounts
+  await sdk.fundAccount({ accountAddress: alice.accountAddress.toString(), amount: ALICE_INITIAL_BALANCE });
+  await sdk.fundAccount({ accountAddress: sponser.accountAddress.toString(), amount: SPONSER_INITIAL_BALANCE });
+
+  // Show account balances
+  const aliceBalanceBefore = await sdk.getAccountCoinsData({ accountAddress: aliceAddres });
+  const sponserBalanceBefore = await sdk.getAccountCoinsData({ accountAddress: sponserAddress });
+
+  console.log("\n=== Balances ===\n");
+  console.log(`Alice's balance is: ${aliceBalanceBefore[0].amount}`);
+  console.log(`Bobs's balance is: ${BOB_INITIAL_BALANCE}`);
+  console.log(`Sponser's balance is: ${sponserBalanceBefore[0].amount}`);
+
+  if (aliceBalanceBefore[0].amount !== ALICE_INITIAL_BALANCE) throw new Error("Alice's balance is incorrect");
+  if (sponserBalanceBefore[0].amount !== SPONSER_INITIAL_BALANCE) throw new Error("Sponsers's balance is incorrect");
+
+  // Generate a fee payer (aka sponser) transaction
+  // with Alice as the sender and Sponser as the fee payer
+  console.log("\n=== Submitting Transaction ===\n");
+  const transaction = await sdk.generateTransaction({
+    sender: aliceAddres,
+    feePayerAddress: sponserAddress,
+    data: {
+      function: "0x1::aptos_account::transfer",
+      type_arguments: [],
+      arguments: [bob.accountAddress, new aptos.U64(TRANSFER_AMOUNT)],
+    },
+  });
+
+  // Alice and Sponser to sign the transaction (order matters)
+  const senderSignature = sdk.signTransaction({ signer: alice, transaction });
+  const sponserSignature = sdk.signTransaction({ signer: sponser, transaction });
+
+  // Submit the transaction to chain
+  const committedTxn = await sdk.submitTransaction({
+    transaction,
+    senderAuthenticator: senderSignature,
+    secondarySignerAuthenticators: { feePayerAuthenticator: sponserSignature },
+  });
+
+  console.log(`Submitted transaction: ${committedTxn.hash}`);
+  await sdk.waitForTransaction({ txnHash: committedTxn.hash });
+
+  console.log("\n=== Balances after transfer ===\n");
+  const aliceBalanceAfter = await sdk.getAccountCoinsData({ accountAddress: aliceAddres });
+  const bobBalanceAfter = await sdk.getAccountCoinsData({ accountAddress: bobAddress });
+  const sponserBalanceAfter = await sdk.getAccountCoinsData({ accountAddress: sponserAddress });
+
+  // Bob should have the transfer amount
+  if (bobBalanceAfter[0].amount !== TRANSFER_AMOUNT) throw new Error("Bob's balance after transfer is incorrect");
+
+  // Alice should have the initial balance minus tranfer amount
+  if (aliceBalanceAfter[0].amount !== ALICE_INITIAL_BALANCE - TRANSFER_AMOUNT)
+    throw new Error("Alice's balance after transfer is incorrect");
+
+  // Sponser should have the initial balance minus gas
+  if (sponserBalanceAfter[0].amount >= SPONSER_INITIAL_BALANCE)
```

Comment:
if we want to be *really* clear, you can also check the actual gas used in the tx and throw an error if it doesn't match

Reply:
interestingly, 
Sponsor initial balance = 100_000_000
Transaction gas used = 1010
Expected sponsor final balance = 99998990
Sponsor final balance fron Indexer = 99899000
Sponsor final balance from fullnode = 99899000

Not sure where the extra 10 is from..?